src/JNI/com/amazonaws/kinesis/video/producer/jni/Parameters.cpp

```diff
@@ -249,10 +251,194 @@ BOOL setClientInfo(JNIEnv *env, jobject clientInfo, PClientInfo pClientInfo) {
         CHK_JVM_EXCEPTION(env);
     }
 
+    methodId = env->GetMethodID(cls, "getMetricLoggingPeriod", "()J");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getMetricLoggingPeriod");
+        pClientInfo->metricLoggingPeriod = 0;
+    } else {
+        pClientInfo->metricLoggingPeriod = env->CallLongMethod(clientInfo, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+
+    methodId = env->GetMethodID(cls, "getReservedCallbackPeriod", "()J");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getReservedCallbackPeriod");
+        pClientInfo->reservedCallbackPeriod = 0;
+    } else {
+        pClientInfo->reservedCallbackPeriod = env->CallLongMethod(clientInfo, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+
+    methodId = env->GetMethodID(cls, "getKvsRetryStrategy", "()Lcom/amazonaws/kinesisvideo/producer/KvsRetryStrategy;");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getKvsRetryStrategy");
+    } else {
+        kvsRetryStrategy = (jobject) env->CallObjectMethod(clientInfo, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+    setKvsRetryStrategy(env, kvsRetryStrategy, &pClientInfo->kvsRetryStrategy);
+    
+    methodId = env->GetMethodID(cls, "getKvsRetryStrategyCallbacks", "()Lcom/amazonaws/kinesisvideo/producer/KvsRetryStrategyCallbacks;");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getKvsRetryStrategyCallbacks");
+    } else {
+        kvsRetryStrategyCallbacks = (jobject) env->CallObjectMethod(clientInfo, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+    setKvsRetryStrategyCallbacks(env, kvsRetryStrategyCallbacks, &pClientInfo->kvsRetryStrategyCallbacks);
+
 CleanUp:
     return STATUS_FAILED(retStatus) ? FALSE : TRUE;
 }
 
+VOID setKvsRetryStrategy(JNIEnv *env, jobject kvsRetryStrategy, PKvsRetryStrategy pKvsRetryStrategy)
+{
+    jmethodID methodId = NULL;
+    jclass cls = NULL;
+    BOOL nullKvsRetryStrategy = FALSE;
+
+    CHECK(env != NULL && pKvsRetryStrategy != NULL);
+
+    if (kvsRetryStrategy == NULL) {
+        nullKvsRetryStrategy = TRUE;
+        goto CleanUp;
+    }
+
+    cls = env->GetObjectClass(kvsRetryStrategy);
+    if (cls == NULL) {
+        DLOGW("Failed to create Java kvsRetryStrategy class.");
+        nullKvsRetryStrategy = TRUE;
+        goto CleanUp;
+    }
+
+    methodId = env->GetMethodID(cls, "getRetryStrategy", "()Lcom/amazonaws/kinesisvideo/producer/RetryStrategy");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getRetryStrategy, setting pRetryStrategy to null.");
+        pKvsRetryStrategy->pRetryStrategy = NULL;
+    } else {
+        if (env->CallLongMethod(kvsRetryStrategy, methodId) == 0) {
```

Comment:
Why isnt this following the same pattern as the rest. We need to check JVM exception and error out. Not silence the errors.